### DISCUSSION TITLE
[Question]Cannot download pyarmor-regfile-****.zip for old-license

### DISCUSSION THREAD

**Discussion by otar0** [dashingsoft/pyarmor#1094] [Q&A] [answered]
We use PyArmor Version 7.4.0.
We want to use old licenses.

We see the following issue.
https://github.com/dashingsoft/pyarmor/issues/985

To register the license, we try the second method in the pyarmor-regcode-xxxx.txt which says we can download pyarmor-regfile-****zip from URL.

When we accessed the URL, there were only the following message. We cannot download the zip file.

"
Pyarmor has been upgraded to 8.0, and there are big changes in Pyarmor 8.0. If never upgrade to 8.0, there is a little change, using .zip file other than .txt file to register Pyarmor, following the second method described in the pyarmor-regcode-xxxx.txt: downloading .zip once, then use this .zip file to register later. Now the download link still works, but it may not work soon, please save and backup .zip file. More information refer to https://github.com/dashingsoft/pyarmor/issues/980
"

In this case, how can we get the pyarmor-regfile-****.zip ?

Note we may have used the `pyarmor register pyarmor-regcode-xxxx.txt ` more than 10 times without the knowing the limitation mentioned in  the follwing.

https://github.com/dashingsoft/pyarmor#important-notes-for-users-of-pyarmor-prior-to-80

Thank you !

**Comment by jondy** [comment]
Just try it after 1 minute, the download link should work. It's broken temporary.

Do not run it more than 3 times in 1 minute, server will reject too many requests with `HTTP 503 error`

**Comment by benoitclem** [comment]
Hi, @jondy.

It might be a little redundant but I got exactly the same issue here. Since you replied juste one hour ago I thought that it might just work now, but still the same error as the OP mentionned.

Should I wait a little bit more as well?

Thanks for you reply.
Clement

**Comment by otar0** [comment]
We just accessed the URL, there were the following message.

<img width="1286" alt="スクリーンショット 2023-04-07 9 24 54" src="https://user-images.githubusercontent.com/130028719/230524986-8389044e-feac-48dd-829d-3cc7e03608c2.png">

So we waited 5 minutes, but the download did not start.
What should we do?

Thank you!

**Comment by jondy** [comment, marked as answer]
@otar0 Sorry, there is a bug in server side. I have fixed it, just try it again. At first time, it will still report this error, but next 2 times, the download link should work. Feel free to let me know if there is still problem.

**Reply by benoitclem** [reply]
It worked, i could download the zip file. 💯

**Reply by otar0** [reply]
Thank you !
We could also download the file.

**Reply by ghost** [reply]
@jondy 

Now I also tried to download pyarmor-regfile-xxxx.zip by accessing the URL in pyarmor-regcode-xxxx.txt to use the old license, but failed.

When I access the URL, "503 Service Temporarily Unavailable" or "exceptions must derive from BaseException" are displayed, and it seems that the zip cannot be downloaded.

Is there any way to download it?

**Reply by jondy** [reply]
@yasunori-doi please send order information to pyarmor@163.com

**Reply by ghost** [reply]
@jondy 
Today, I accessed the URL to download pyarmor-regfile-xxxx.zip again and was able to download the zip without any problems.
